heron/executor/src/python/heron_executor.py

```diff
@@ -192,6 +201,50 @@ def __init__(self, args, shell_env):
 
     self.state_managers = []
 
+  @staticmethod
+  def parse_args(args):
+    """Uses python argparse to collect positional args"""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("shard", type=int)
+    parser.add_argument("topology_name")
+    parser.add_argument("topology_id")
+    parser.add_argument("topology_defn_file")
+    parser.add_argument("zknode")
+    parser.add_argument("zkroot")
+    parser.add_argument("tmaster_binary")
+    parser.add_argument("stmgr_binary")
+    parser.add_argument("metricsmgr_classpath")
+    parser.add_argument("instance_jvm_opts")
+    parser.add_argument("classpath")
+    parser.add_argument("master_port")
+    parser.add_argument("tmaster_controller_port")
+    parser.add_argument("tmaster_stats_port")
+    parser.add_argument("heron_internals_config_file")
+    parser.add_argument("component_rammap")
+    parser.add_argument("component_jvm_opts_in_base64")
+    parser.add_argument("pkg_type")
+    parser.add_argument("topology_bin_file")
+    parser.add_argument("heron_java_home")
+    parser.add_argument("shell_port")
+    parser.add_argument("heron_shell_binary")
+    parser.add_argument("metricsmgr_port")
+    parser.add_argument("cluster")
+    parser.add_argument("role")
+    parser.add_argument("environ")
+    parser.add_argument("instance_classpath")
+    parser.add_argument("metrics_sinks_config_file")
+    parser.add_argument("scheduler_classpath")
+    parser.add_argument("scheduler_port")
+    parser.add_argument("python_instance_binary")
+
+    return parser.parse_args(args[1:])
```

Comment:
Since you removed [this part](https://github.com/twitter/heron/pull/1493/files#diff-2a669e084143b86b5766a49245f8e357L650), can you use `parse_known_args` as in [heron/tools/explorer/src/python/main.py#L172](https://github.com/twitter/heron/blob/12425366a9ade8501675c1e29c2ea34cb36044cb/heron/tools/explorer/src/python/main.py#L172), and check if there is any unknown args?